sql/mysqld.cc

```diff
@@ -4781,6 +4781,7 @@ int mysqld_main(int argc, char **argv)
   sys_var_init();
   ulong requested_open_files;
   adjust_related_options(&requested_open_files);
+  my_init_signals();
```

Comment:
The change looks correct by code review, but how come SIGTERM does not kill a server with a PFS thread running then currently? Or it does? What is the relation between this change and the current bug?

Reply:
Sorry, I have not made myself clear about this. The problem is that after fixing this problem, an other crash happened - due to different reason. I have posted a comment in the bug description. Please have a look and tell me if this make more sense now.

Reply:
This is due to the fact that error handling thread (function start_signal_handler) is called long after threadpool gets initialized. Basing on your comment in code - theadpool initialization is heavy which means that PFS threads have to survive longer period of time without any signal handling - as they do not get signal mask set.